test/unit/workDiv/src/FooVec.hpp

```diff
@@ -0,0 +1,152 @@
+/* Copyright 2022 Sergei Bastrakov, Jan Stephan, Mehmet Yusufoglu
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+#pragma once
+
+#include <alpaka/dim/Traits.hpp>
+#include <alpaka/elem/Traits.hpp>
+#include <alpaka/extent/Traits.hpp>
+#include <alpaka/vec/Vec.hpp>
+
+#include <array>
+#include <iostream>
+
+//! User defined vector for testing the usability of any vector type.
+//!
+//! \tparam TVec The data type.
+//! \tparam Size Vector size as a non-type parameter.
+template<typename TVec, std::size_t Size>
+class FooVec
+{
+public:
+    static_assert(Size <= 3, "Size must be 3 or smaller");
+    std::array<TVec, Size> arr;
+
+    // Default Constructor
+    FooVec()
+    {
+        arr.fill(TVec());
+    }
+
+    // Constructor with initializer list
+    FooVec(std::initializer_list<TVec> initList)
+    {
+        if(initList.size() <= Size)
+        {
+            std::copy(initList.begin(), initList.end(), arr.begin());
+        }
+        else
+        {
+            throw std::out_of_range("Initializer list size exceeds array size");
+        }
+    }
+
+    // Destructor
+    ~FooVec()
+    {
+        // No explicit cleanup required for this example
+    }
+
+    // Copy Constructor
+    FooVec(FooVec const& other)
+    {
+        arr = other.arr;
+    }
+
+    // Copy Assignment Operator
+    FooVec& operator=(FooVec const& other)
+    {
+        if(this != &other)
+        {
+            arr = other.arr;
+        }
+        return *this;
+    }
+
+    // Move Assignment Operator
+    FooVec& operator=(FooVec&& other) noexcept
+    {
+        if(this != &other)
+        {
+            arr = std::move(other.arr);
+        }
+        return *this;
+    }
+
+    // Example member function to print the contents of the array
+    void printArray() const
+    {
+        for(auto const& element : arr)
+        {
+            std::cout << element << ' ';
+        }
+        std::cout << std::endl;
+    }
+};
+
+namespace alpaka::trait
+{
+    //! Traits to separate the user defined example vector FooVec from the other structures used in WorkDivMembers
+    //! constructor
+    template<typename T, typename = void>
+    struct has_member_arr : std::false_type
+    {
+    };
+
+    template<typename T>
+    struct has_member_arr<T, std::void_t<decltype(&T::arr)>> : std::true_type
+    {
+    };
+
+    //! The DimType specialization for the user defined vector which has a member called "arr"
+    //! \tparam T The vector type
+    template<typename T>
+    struct DimType<T, std::enable_if_t<has_member_arr<T>::value>>
```

Comment:
```suggestion
    //! The DimType specialization for the user defined vector which has a member called "arr"
    //! \tparam T The vector type
    template<typename T>
    struct DimType<FooVec>
```
Why don't you just specialize for `FooVec` directly? I don't see a need for SFINAE here.

Applies to other traits as well.

Reply:
FooVec is a template, not a type. But i will try to formulate your approach.

Reply:
Sorry, my bad. Should be:
```c++
template<typename TVec, std::size_t Size>
struct DimType<FooVec<TVec, Size>>
```

Reply:
done thanks.